src/main/java/seedu/kitchenhelper/object/Expenditure.java

```diff
@@ -0,0 +1,169 @@
+        } else {
+            pastMonday = anyMonday;
+        }
+        if (lastSavedDate == null || lastSavedDate.before(pastMonday)) {
+            totalExpenditure = 0;
+            amountUsedInCooking = 0;
+        }
+    }
+
+    public void addToExpenditure(double price, int quantity) {
+        totalExpenditure += (price * quantity);
+        Storage.saveExpenditureData();
+    }
+
+    public void removeFromExpenditure(Ingredient ingredientToDelete, Integer quantityToDelete) {
+        String userResponse = promptUser(PROMPT_REMOVE_FROM_EXPENDITURE);
+        if (userResponse.equalsIgnoreCase("yes")) {
+            double price = changePrice(ingredientToDelete, quantityToDelete);
+            totalExpenditure -= price;
+            System.out.println(String.format((REMOVAL_SUCCESS), price));
+        } else {
+            System.out.println(NO_CHANGE);
+        }
+    }
+
+    public void addToAmountUsed(Ingredient ingredientToDelete, Integer quantityToDelete) {
+        String userResponse = promptUser(PROMPT_ADD_TO_AMOUNT_USED);
+        if (userResponse.equalsIgnoreCase("yes")) {
+            addAmountForCooking(ingredientToDelete, quantityToDelete);
+        } else {
+            System.out.println(NO_CHANGE);
+        }
+    }
+
+    public void addAmountForCooking(Ingredient ingredientToDelete, Integer quantityToDelete) {
+        amountUsedInCooking += changePrice(ingredientToDelete, quantityToDelete);
+        Storage.saveExpenditureData();
+    }
+
+    public void editExpenditure(Ingredient ingredientToDelete, Integer quantityToDelete) {
+        removeFromExpenditure(ingredientToDelete, quantityToDelete);
+        addToAmountUsed(ingredientToDelete, quantityToDelete);
+        Storage.saveExpenditureData();
+    }
+
+    public double changePrice(Ingredient ingredientToDelete, Integer quantityToDelete) {
+        double pricePerPiece = ingredientToDelete.getPrice();
+        int totalQuantityOfItem = ingredientToDelete.getQuantity();
+        if (quantityToDelete == null) {
+            return pricePerPiece * totalQuantityOfItem;
+        } else {
+            return pricePerPiece * quantityToDelete;
+        }
+    }
+
+    public String promptUser(String prompt) {
+        String userResponse;
+        System.out.println(prompt);
+        try {
+            userResponse = new Scanner(System.in).nextLine().trim();
+        } catch (NoSuchElementException e) {
+            return "no";
+        }
+        while (!isValidResponse(userResponse)) {
+            System.out.println("Please enter either \"Yes\"/\"No\"\n" + Ui.DIVIDER);
+            userResponse = new Scanner(System.in).nextLine().trim();
+        }
+        return userResponse;
+    }
+
+    public boolean isValidResponse(String userResponse) {
+        String response = userResponse;
+        if (response.equalsIgnoreCase("no") || response.equalsIgnoreCase("yes")) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
Do check all these functions would be used for the code later on. 
If unused, remove them in the next PR, along side with the other methods in this Expenditure class.